dcos/marathon.py

```diff
@@ -290,7 +290,10 @@ def add_app(self, app_resource):
         else:
             app_json = app_resource
 
-        response = _http_req(http.post, url,
+        if not isinstance(app_json, list):
+            app_json = [app_json]
+
+        response = _http_req(http.put, url,
```

Comment:
I talked with @aquamatthias and the /v2/apps endpoint supports only single apps via the POST method but single as well as multiple apps via the PUT method. When sending a POST you get the complete app definition returned while when using PUT you get the deployment ID returned. I checked the code base and it seems we're not currently doing anything with the return value. When talking with the Marathon team they said that the deployment ID might be useful for implementing a blocking function that waits for the deployment to finish after sending a app definition. This would allow user stories where they can use dcos-cli in their scripts and perform steps like 1) start database server, 2) create database schema, which is currently harder to implement in simple bash.